Reflect changes in chain-extensions API (#142)
paritytech/ink-docs: versioned_docs/version-4.0.0-alpha.1/macros-attributes/chain-extension.md

## Attributes

> There are 2 different attributes with which the chain extension methods

Reviewer: ```suggestion
There are two different attributes with which the chain extension methods
```

This is a better written style.

Default value: `true`

> By default all chain extension methods should return a `Result<T, E>` where `E: From<Self::ErrorCode>`.

Reviewer: Can you clarify what the implication is if they don't?

Author: I think lines 106-107 states the implication

Author: > Can you clarify what the implication is if they don't?

I think lines 106-107 states the implication

will always indicate success. Therefore it will always load and decode the output buffer and loses
the `E: From<Self::ErrorCode` constraint for the call.

> Note that if a chain extension method does not return `Result<T, E>` where `E: From<Self::ErrorCode>`

Reviewer: ```suggestion
Note that if a chain extension method does not return `Result<T, E>` where `E: From<Self::ErrorCode>`,
```

but `handle_status = true` it will still return a value of type `Result<T, Self::ErrorCode>`.

## Usage: `handle_status` + `Result<T, E>` return type

Use both `handle_status = false` and non-`Result` return type for the same chain extension method
if a call to it may never fail and never returns a `Result` type.

## Combinations

> Due to the possibility to flag a chain extension method with `handle_status` and return or not `Result<T, E>`

Reviewer: ```suggestion
Due to the possibility to flag a chain extension method with `handle_status` and either (1) return `Result<T, E>`
or (2) return just `T`
```

If I understood it correctly.
